File: Python/Programmers/TargetNumber.py
def my_dfs(numbers, cur_idx, cur_value, target):
    if cur_idx == len(numbers) - 1:
        if cur_value == target:
            return 1
        else:
            return 0
    
    return my_dfs(numbers, cur_idx + 1, cur_value + numbers[cur_idx + 1], target) + my_dfs(numbers, cur_idx + 1, cur_value - numbers[cur_idx + 1], target)
    
    
def solution(numbers, target):
    # x1, x-1
    stack = []
    cur_idx = 0
    total = my_dfs(numbers, cur_idx, numbers[cur_idx], target) + my_dfs(numbers, cur_idx, -numbers[cur_idx], target)
    return total

File: Java/Programmers/TargetNumber.java
class Solution {
    static int N;
    static int answer = 0;
    static int globalTarget = 0;
    
    void dfs(int[] numbers, int total, int operCnt) {
        if (operCnt == N) {
            if (total == globalTarget) {
                answer += 1;
            }
            return;
        }
        
        for (int i = 0; i < 2; i++) {
            char operation = "+-".charAt(i);
            int tmpTotal = 0;
            if (operation == '+') {
                tmpTotal = total + numbers[operCnt];
            } else if (operation == '-') {
                tmpTotal = total - numbers[operCnt];
            }
            dfs(numbers, tmpTotal, operCnt + 1);
        }
    }
    
    public int solution(int[] numbers, int target) {
        N = numbers.length;
        globalTarget = target;
        dfs(numbers, 0, 0);
        return answer;
    }
}